### ISSUE TITLE
Battery and Powerline-multiline conflict

### ISSUE THREAD

**Issue by aarmn** [ohmybash/oh-my-bash#193]
[](url)
![image](https://user-images.githubusercontent.com/29726901/103443252-8a95fc00-4c72-11eb-8e3d-4614b1c05055.png)
when battery plugin is enabled this happen

**Comment by Delta456** [comment]
How is the battery plugin enabled?

**Comment by agreeff** [comment]
I get the same behaviour with this combination.. I don't *intentionally* use my GP73 on battery power very often, but I do sometimes plug it in and forget to turn on the plug. having battery percentage in my terminal (aka: my second home) really does help, but the extra line is more than I little annoying.

can't seem to reproduce this with other "multiline" themes though, but the `battery` plugin doesn't seem to add anything on the other themes I've tried. I also tried removing all other plugins, and even completions and aliases, just in case; still get the same behaviour.

**Comment by JaydevSR** [comment]
I am also getting a similar a behavior with battery and powerline-multiline together, I did a fresh install no customizations from my side.
![Screenshot from 2021-02-17 01-50-54](https://user-images.githubusercontent.com/59474411/108117566-1220b880-70c3-11eb-8f18-209347c525f0.png)

**Comment by KrappRamiro** [comment]
Same issue on my side, clean installation with powerline-multiline and battery plugin enabled

**Comment by akinomyoga** [comment]
No one seems to describe what is the issue explicitly, but you are talking about the extra line between the line showing information (times, batteries, etc.) and the line where the user command will be input, right?

**Comment by aarmn** [comment]
> No one seems to describe what is the issue explicitly, but you are talking about the extra line between the line showing information (times, batteries, etc.) and the line where the user command will be input, right?

Yup, that extra blue line was the problem. But at this point this issue is so old that I do not use oh-my-bash, and my machine os changed multiple times, therefore I cannot share a MWE, but I had some similar issues with starship as well, but way more rarely, so I think it's a general problem with all these codes, If I had that I would have been able to check it on different terminals, but I don't think it only happens on yakuake

**Comment by akinomyoga** [comment]
Thank you for the information!

> I would have been able to check it on different terminals, but I don't think it only happened on yakuake, but I don't know,

OK, so you have been using yakuake at that time? I have now tried `konsole`, `lxterminal`, `termit`, `terminology`, `terminator`, `xfce4-terminal`, `xterm`, `rxvt`, `urxvt256c`, `alacritty`, `mintty`, and `kitty`, but I couldn't reproduce the problem in any of these terminals. I also tried `mlterm` with which the rendering is broken even without battery, but this is because mlterm uses a different width mode of Unicode `East_Asian_Width` informative property. Maybe I should set up `yakuake` later.

> maybe it was a font problem

Yeah, probably it is related to the calculation of the width of each character. Some terminals uses the font metrics to determine the character width, but the basic agreement between most terminals and terminal applications are to use the system's `wcwidth`. Even with this rule, there will be problems when the user log into the host from the remote terminal because the behavior of `wcwidth` depends on the system (with different combinations of Unicode versions, Unicode Emoji versions, and support for the emoji sequences and the level of grapheme clusters such as "legacy" vs "extended", etc.).

> or a bug fixed by now!

The original problem might be solved in the latest version of yakuake, but @KrappRamiro seems to have experienced the same now, though I'm not sure if @KrappRamiro uses the same terminal. Hi, @KrappRamiro, what is your terminal and the font used there?

**Comment by akinomyoga** [comment]
I now tried `yakuake` in Ubuntu 20.04 LTS and confirmed that the problem is still present there! In addition, I could also reproduce the problem in `gnome-terminal`, `lxterminal`, and `kitty` in Ubuntu 20.04 LTS even though they did not exhibit the problem in my default host, Fedora 36 Server Edition. It seems to be related the fonts installed on the system.

I'm not sure if this is something that can be fixed at the oh-my-bash side but will later take a look at it.

**Comment by aarmn** [comment]
I think the best we can do is to pinpoint the problem, because foremost, it's not "just omb" problem, then we can at least warn users in README about such problems happening again. But first we clarify its fonts? Terminals? Even I am not really sure at this point
An effective solution can later be copied by many other projects as well!

**Comment by KrappRamiro** [comment]
> Hi, @KrappRamiro, what is your terminal and the font used there?
Hi, sorry for slow replying, my terminal is gnome-terminal, and the font is SauceCodePro Nerd Font.

But for some weird reason, the bug its not happening anymore. I dont know why, maybe its because the emoji isnt appearing, so its no longer pushing the blue square out of line?

**Comment by akinomyoga** [comment]
Thank you for the information! So you haven't changed the font since you last reported it? It is interesting that even with the same font, the problem may happen or not happen. Maybe there are even other factors.

**Comment by KrappRamiro** [comment]
I haven't changed the font, yeah.
But recently, i was able to reproduce the bug, and it only seems to happen when the ⚡ emoji appears, and that is when battery is full and AC is connected. So the bug seems to be related to the use of the ⚡ emoji

**Comment by dankamongmen** [comment]
> **Edit**: I found `notcurses-info` as a terminfo replacement command, but its output format doesn't seem to be shell-script friendly like that of `tput`.

yeah, i focused on this rather than tput-like functionality. i don't think i have anything to help you immediately, but keep me in the loop!

**Comment by akinomyoga** [comment]
Thank you for your reply!

**Comment by alsimoneau** [comment]
I fixed it by replacing the icon in `/usr/share/oh-my-bash/themes/powerline-multiline/powerline-multiline.theme.sh` at line 41 with the nerdfont character 󱐋 (f140b).
I also used the nerdfont characters for the Python and Ruby symbols while at it.
